cumulus/parachains/runtimes/assets/test-utils/src/test_cases.rs

```diff
@@ -1339,3 +1334,188 @@ macro_rules! include_create_and_manage_foreign_assets_for_local_consensus_parach
 		}
 	}
 );
+
+/// Test-case makes sure that `Runtime` can reserve-transfer asset to other parachains
+pub fn reserve_transfer_native_asset_works<
+	Runtime,
+	AllPalletsWithoutSystem,
+	XcmConfig,
+	HrmpChannelOpener,
+	HrmpChannelSource,
+	LocationToAccountId,
+>(
+	collator_session_keys: CollatorSessionKeys<Runtime>,
+	existential_deposit: BalanceOf<Runtime>,
+	alice_account: AccountIdOf<Runtime>,
+	unwrap_pallet_xcm_event: Box<dyn Fn(Vec<u8>) -> Option<pallet_xcm::Event<Runtime>>>,
+	unwrap_xcmp_queue_event: Box<
+		dyn Fn(Vec<u8>) -> Option<cumulus_pallet_xcmp_queue::Event<Runtime>>,
+	>,
+	weight_limit: WeightLimit,
+) where
+	Runtime: frame_system::Config
+		+ pallet_balances::Config
+		+ pallet_session::Config
+		+ pallet_xcm::Config
+		+ parachain_info::Config
+		+ pallet_collator_selection::Config
+		+ cumulus_pallet_parachain_system::Config
+		+ cumulus_pallet_xcmp_queue::Config,
+	AllPalletsWithoutSystem:
+		OnInitialize<BlockNumberFor<Runtime>> + OnFinalize<BlockNumberFor<Runtime>>,
+	AccountIdOf<Runtime>: Into<[u8; 32]>,
+	ValidatorIdOf<Runtime>: From<AccountIdOf<Runtime>>,
+	BalanceOf<Runtime>: From<Balance>,
+	<Runtime as pallet_balances::Config>::Balance: From<Balance> + Into<u128>,
+	XcmConfig: xcm_executor::Config,
+	LocationToAccountId: ConvertLocation<AccountIdOf<Runtime>>,
+	<Runtime as frame_system::Config>::AccountId:
+		Into<<<Runtime as frame_system::Config>::RuntimeOrigin as OriginTrait>::AccountId>,
+	<<Runtime as frame_system::Config>::Lookup as StaticLookup>::Source:
+		From<<Runtime as frame_system::Config>::AccountId>,
+	<Runtime as frame_system::Config>::AccountId: From<AccountId>,
+	HrmpChannelOpener: frame_support::inherent::ProvideInherent<
+		Call = cumulus_pallet_parachain_system::Call<Runtime>,
+	>,
+	HrmpChannelSource: XcmpMessageSource,
+{
+	let runtime_para_id = 1000;
+	ExtBuilder::<Runtime>::default()
+		.with_collators(collator_session_keys.collators())
+		.with_session_keys(collator_session_keys.session_keys())
+		.with_tracing()
+		.with_safe_xcm_version(3)
+		.with_para_id(runtime_para_id.into())
+		.build()
+		.execute_with(|| {
+			let mut alice = [0u8; 32];
+			alice[0] = 1;
+			let included_head = RuntimeHelper::<Runtime, AllPalletsWithoutSystem>::run_to_block(
+				2,
+				AccountId::from(alice).into(),
+			);
+
+			// reserve-transfer native asset with local reserve to remote parachain (1234)
+
+			let other_para_id = 1234;
+			let native_asset = MultiLocation::parent();
+			let dest = MultiLocation::new(1, X1(Parachain(other_para_id)));
+			let dest_beneficiary = MultiLocation::new(1, X1(Parachain(other_para_id)))
```

Comment:
just a nit I realized now, `dest_beneficiary` should be relative to the `dest`, so better:
```
MultiLocation {
    parents: 0,
    interior: X1(
        AccountId32 {
	    network: None,
	    id: sp_runtime::AccountId32::new([3; 32]).into(),
        })
}
```
We should also fix it [everywhere e.g.](https://github.com/paritytech/polkadot-sdk/pull/1215/files#diff-7b47f9a0d27321e248470a1a31486322dbc3bab296aab6687fde4e979b34a248R44)

Anyway, it is not relevant for this unit-tests because we dont execute target chain stuff (but with xcm-executor we can).

Reply:
done